clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -5209,6 +5220,92 @@ static void handleIntelFPGAMergeAttr(Sema &S, Decl *D, const ParsedAttr &AL) {
                  IntelFPGAMergeAttr(S.Context, AL, Results[0], Results[1]));
 }
 
+/// Handle the bank_bits attribute.
+/// This attribute accepts a list of values greater than zero.
+/// This is incompatible with the register attribute.
+/// The numbanks and bank_bits attributes are related. If numbanks exists
+/// when handling bank_bits they are checked for consistency. If numbanks
+/// hasn't been added yet an implicit one is added with the correct value.
+/// If the user later adds a numbanks attribute the implicit one is removed.
+/// The values must be consecutive values (i.e. 3,4,5 or 2,1).
+static void handleIntelFPGABankBitsAttr(Sema &S, Decl *D,
+                                        const ParsedAttr &Attr) {
+  checkForDuplicateAttribute<IntelFPGABankBitsAttr>(S, D, Attr);
```

Comment:
Add D decl validity check:
`
  if (D->isInvalidDecl())
    return;
`

Reply:
Are you sure we need this check? It seems that other attributes handling has no such check. I think that it is handled higher in the call stack.

Reply:
It depends on a contract we have with clang code:
1. We trust function inputs;
2. We don't trust function inputs.

Reply:
I don't have here a strong evidences to push anyone to follow my own preferences, but as far as I'm a pedantic person - I'd prefer to add this check at least in a code what I write.

Reply:
Quickly looking into the code I did not find such checks.

Reply:
I mean declaration invalidation checks before attribute handling earlier in the code.